docs/index.md

```diff
@@ -0,0 +1,85 @@
+lead: "The Filecoin Virtual Machine (FVM) is an interface that developers can use to deploy smart-contracts on the Filecoin network. FVM contracts will introduce the ability to perfom on-chain computation, or the computation of a state. This is the perfect pairing of blockchain storage and computation that Web3 has been waiting for."
+---
+
+# The Filecoin Virtual Machine
+
+The Filecoin Virtual Machine (FVM) is an interface that enables developers to deploy smart contracts on the Filecoin blockchain. FVM support for smart contracts provides the ability to perform on-chain state computations that guide storage deals from pre-commit through committed states. This is the perfect pairing of blockchain storage and computation that Web3 has been waiting for.
+
+## Why the FVM is useful
+
+There has been a huge demand from the developer community for compatibility with Ethereum and Solidity out of the box. With the FVM, developers can access the massive corpus of audited and battle-tested smart contracts written in Solidity for the Ethereum Virtual Machnine (EVM). The FVM also supports other foreign runtimes and virtual machines, such as JS/SES, LLVm, IR, and BPF.
+
+The FVM is flexible. Because it's not tightly-coupled to Filecoin, developers can adapt to it in other contexts, such as the Interplanetary File System (IPFS) and Interplanetary Linked Data (IPLD).
+
+The FVM is easy for developers who know IPLD to adapt to. The state tree and actor states that the FVM uses to guide storage deals from pre-commit through committed states are IPLD objects.
+
+## Vision and goals
+
+To mitigate the well-known risks of compiling Solidity used by the EVM, we're not working directly in Solidity. Instead, we're building the reference SDK in Rust, because it produces very succinct WASM code and keeps runtime costs down.
+
+Because the FVM essentially executes computation on IPLD inputs to produce outputs, building libraries and primitives to interact with IPLD data is a core priority.
+
+Because user-defined actors will exponentially increase demand for space on the blockchain, we're looking into solutions, such as sharding (a kind of partitioning) and hierarchical consensus with parallel execution, both of which support speedier processing.
+
+## Use-cases
+
+Here are some use cases that are possible with the FVM. With centralized data centers, a lot of these use cases were incredibly hard, if not impossible. With decentralization and Layer 1 programmability (computation for state changes), we're unleashing boundless opportunities.
+
+### Decentralized, verifiable computation
+
+The FVM enables both computation and storage under a single roof. Imagine being able to apply a computation on data, incentivizing its execution all the way to the end, and certifying that the result AND the output are verifiably correct.
```

Comment:
> to apply a computation on data

Is this grammatically correct?

```suggestion
The FVM enables both computation and storage under a single roof. Imagine being able compute data on-chain and certify that the result and the output are verifiably correct.
```

Reply:
@johnnymatthews Typos.... And I'll take out the "imagine" too. Also gimmicky.